File: Longest Consecutive Sequence/Longest_Consecutive_Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        map<int,int> mpp;
        for(auto i: nums) {
            mpp[i] = 1;
        }
        int ans = 0;
        for(auto &i: mpp) {
            i.second = max(i.second,mpp[i.first-1]+1);
            ans = max(ans,i.second);
        }
        return ans;
    }
};

File: Longest Consecutive Sequence/Longest_Consecutive_Sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Map<Integer,Integer> mpp = new TreeMap<>();
        for(int i: nums) {
            mpp.put(i,1);
        }
        int ans = 0;
        for(int i : mpp.keySet()) {
            if(mpp.containsKey(i-1)) {
                mpp.put(i,mpp.get(i-1)+1);
            }
            ans = Math.max(ans,mpp.get(i));
        }
        return ans;
    }
}